File: Map/src/app/shared/DanhMucChucNang.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { environment } from 'src/environments/environment';
import { DanhMucChucNang } from 'src/app/shared/DanhMucChucNang.model';
import { BaseService } from './Base.service';
@Injectable({
    providedIn: 'root'
})
export class DanhMucChucNangService extends BaseService{

    ListChild: DanhMucChucNang[] | undefined;
    ListParent: DanhMucChucNang[] | undefined;

    DisplayColumns001: string[] = ['ID', 'ParentID', 'Name', 'Code', 'Display', 'SortOrder', 'Active', 'Save'];
    
    constructor(public httpClient: HttpClient) {
        super(httpClient);
        this.Controller = "DanhMucChucNang";
    }

    GetSQLByThanhVienID_ActiveToListAsync() {
        var lastUpdatedMembershipID = localStorage.getItem(environment.ThanhVienID);
        if (lastUpdatedMembershipID) {
            this.BaseParameter.ThanhVienID = Number(lastUpdatedMembershipID);
        }
        this.BaseParameter.Active = true;
        let url = this.APIURL + this.Controller + '/GetSQLByThanhVienID_ActiveToListAsync';
        const formUpload: FormData = new FormData();
        formUpload.append('data', JSON.stringify(this.BaseParameter));
        return this.httpClient.post(url, formUpload, { headers: this.Headers });
    }
}

File: Service/Implement/DanhMucChucNangService.cs
namespace Service.Implement
{
    public class DanhMucChucNangService : BaseService<DanhMucChucNang, IDanhMucChucNangRepository>
    , IDanhMucChucNangService
    {
        private readonly IDanhMucChucNangRepository _DanhMucChucNangRepository;
        public DanhMucChucNangService(IDanhMucChucNangRepository DanhMucChucNangRepository) : base(DanhMucChucNangRepository)
        {
            _DanhMucChucNangRepository = DanhMucChucNangRepository;
        }
        public override void Initialization(DanhMucChucNang model)
        {
            BaseInitialization(model);
            if (string.IsNullOrEmpty(model.Code))
            {
                model.Code = "#";
            }
        }
        public virtual async Task<List<DanhMucChucNang>> GetSQLByThanhVienID_ActiveToListAsync(long thanhVienID, bool active)
        {
            List<DanhMucChucNang> result = new List<DanhMucChucNang>();
            SqlParameter[] parameters =
            {
                    new SqlParameter("@ThanhVienID",thanhVienID),
                    new SqlParameter("@Active",active),
            };
            result = await GetByStoredProcedureToListAsync("sp_DanhMucChucNangSelectItemsByThanhVienID_Active", parameters);
            return result;
        }
    }
}

